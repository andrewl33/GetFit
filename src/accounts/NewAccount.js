import React, { Component } from 'react';
import { StyleSheet, Text, TextInput, View, Button, Alert } from 'react-native';
import { _addNewUser, _getAllUsers } from '../Storage/UserStorage';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    padding: 10,
    borderRadius: 2,
    justifyContent: 'flex-start',
    alignItems: 'center',
  },
  header: {
    marginTop: 50,
    padding: 2,
  },
  text: {
    fontSize: 20,
  },
});

class Header extends Component {
  render() {
    return (
      <View style={styles.header}>
        <Text style={{ fontSize: 35 }}>New Account</Text>
      </View>
    );
  }
}

export default class NewAccount extends Component {
  constructor(props) {
    super(props);
    this.state = {
      firstName: '',
      lastName: '',
      email: '',
      password: '',
    };
    this.submit = this.submit.bind(this);
  }

  async submit() {
    const { firstName, lastName, email, password } = this.state;

    Alert.alert('Welcome to GetFit!');
    console.log(
      ` firstName = ${firstName}\n`,
      `lastName  = ${lastName}\n`,
      `email     = ${email}\n`,
      `password  = ${password}`
    );

    await _addNewUser(this.state);
    await console.log(await _getAllUsers());
  }

  render() {
    return (
      <View style={styles.container}>
        <Header />
        <View style={{ flexDirection: 'row' }}>
          <Text style={styles.text}>First Name: </Text>
          <TextInput
            style={styles.text}
            onChangeText={input => this.setState({ firstName: input })}
            placeholder="First Name"
          />
        </View>
        <View style={{ flexDirection: 'row' }}>
          <Text style={styles.text}>Last Name: </Text>
          <TextInput
            style={styles.text}
            onChangeText={input => this.setState({ lastName: input })}
            placeholder="Last Name"
          />
        </View>
        <View style={{ flexDirection: 'row' }}>
          <Text style={styles.text}>Email: </Text>
          <TextInput
            style={styles.text}
            onChangeText={input => this.setState({ email: input })}
            placeholder="Email"
          />
        </View>
        <View style={{ flexDirection: 'row' }}>
          <Text style={styles.text}>Password: </Text>
          <TextInput
            style={styles.text}
            onChangeText={input => this.setState({ password: input })}
            placeholder="Password"
            secureTextEntry
          />
        </View>
        <Button onPress={this.submit} title="Create Account" color="#800080" />
      </View>
    );
  }
}
