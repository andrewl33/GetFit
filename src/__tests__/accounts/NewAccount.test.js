import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import NewAccount from '../../accounts/NewAccount';

describe('NewAccount', () => {
  it('renders', () => {
    const tree = renderer.create(<NewAccount />);
    expect(tree).toMatchSnapshot();
  });
});
