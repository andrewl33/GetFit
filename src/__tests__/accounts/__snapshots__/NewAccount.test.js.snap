// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewAccount renders 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "borderRadius": 2,
      "flex": 1,
      "justifyContent": "flex-start",
      "padding": 10,
    }
  }
>
  <View
    style={
      Object {
        "marginTop": 50,
        "padding": 2,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 35,
        }
      }
    >
      New Account
    </Text>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 20,
        }
      }
    >
      First Name: 
    </Text>
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="First Name"
      style={
        Object {
          "fontSize": 20,
        }
      }
      underlineColorAndroid="transparent"
    />
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 20,
        }
      }
    >
      Last Name: 
    </Text>
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Last Name"
      style={
        Object {
          "fontSize": 20,
        }
      }
      underlineColorAndroid="transparent"
    />
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 20,
        }
      }
    >
      Email: 
    </Text>
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Email"
      style={
        Object {
          "fontSize": 20,
        }
      }
      underlineColorAndroid="transparent"
    />
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 20,
        }
      }
    >
      Password: 
    </Text>
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Password"
      secureTextEntry={true}
      style={
        Object {
          "fontSize": 20,
        }
      }
      underlineColorAndroid="transparent"
    />
  </View>
  <View
    accessibilityRole="button"
    accessibilityStates={Array []}
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
            Object {
              "color": "#800080",
            },
          ]
        }
      >
        Create Account
      </Text>
    </View>
  </View>
</View>
`;
